subsys/bluetooth/controller/ll_sw/ll_filter.c

```diff
@@ -624,10 +624,14 @@ void ll_rl_rpa_update(bool timeout)
 		if ((rl[i].taken) && (all || !rl[i].rpas_ready)) {
 
 			if (rl[i].pirk) {
-				err = bt_rpa_create(peer_irks[rl[i].pirk_idx],
-						    &rl[i].peer_rpa);
+				u8_t irk[16];
+
+				sys_memcpy_swap(irk, peer_irks[rl[i].pirk_idx],
+						16);
+				err = bt_rpa_create(irk, &rl[i].peer_rpa);
```

Comment:
Please have a bt_rpa_create_be instead that takes peer_rpa in big-endian. Saves on 2 times as h/w takes BE.